File: internal/entity/atm.go
package entity

import (
    _ "gorm.io/gorm"
    "time"
)

type Atm struct {
    ID            int64 `gorm:"primaryKey"`
    Address       string
    Latitude      float32
    Longitude     float32
    WorkTimeStart time.Time `gorm:"type:time;not null"`
    WorkTimeEnd   time.Time `gorm:"type:time;not null"`
	Services []Service `gorm:"many2many:atms_services;foreignKey:AtmID;joinForeignKey:AtmID;References:ID;joinReferences:ServiceID"`
}

func (a *Atm) TableName() string {
    return "atms"
}

File: src/main/java/com/vtb/hack/hack/entity/Atm.java
package com.vtb.hack.hack.entity;

import java.sql.Time;
import java.util.HashSet;
import java.util.Set;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@EqualsAndHashCode(callSuper = false)
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "atms")
@Data
public class Atm extends BaseEntity {
    private String address;

    @Column(nullable = false)
    private float latitude;

    @Column(nullable = false)
    private float longitude;

    @Column(nullable = false)
    private Time workTimeStart;

    @Column(nullable = false)
    private Time workTimeEnd;

    @EqualsAndHashCode.Exclude
    @Builder.Default
    @ManyToMany(cascade = { CascadeType.PERSIST, CascadeType.MERGE })
    @JoinTable(name = "atms_services", joinColumns = @JoinColumn(name = "atm_id"), inverseJoinColumns = @JoinColumn(name = "service_id"))
    private Set<Service> services = new HashSet<>();
}
